vault/src/system.rs

```diff
@@ -548,13 +552,16 @@ impl VaultService {
 
     async fn maybe_register_vault(&self) -> Result<(), Error> {
         let account_id = self.btc_parachain.get_account_id();
-        let collateral_currency = match self.config.currency_id {
-            Some(x) => x,
-            None => {
-                tracing::info!("Not registering vault -- currency-id not configured");
-                return Ok(());
-            }
-        };
+        let (collateral_currency, _wrapped_currency) =
+            match (self.config.collateral_currency_id, self.config.wrapped_currency_id) {
+                (Some(x), Some(y)) => (x, y),
+                _ => {
+                    tracing::info!(
+                    "Not registering vault -- collateral-currency-id and wrapped-currency-id must both be configured"
```

Comment:
with the change requested above, this would become `"Not registering vault -- collateral-currency-id and wrapped-currency-id not configured"`. Note that we don't require the currency to be set currently - if none is specified, we just don't register. If the user then registers the vault through the ui, the client will pick it up